lib/ansible/modules/validate_argument_spec.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: validate_argument_spec
+short_description: Validate role argument specs.
+description:
+     - This module validates role arguments with a defined argument specification.
+version_added: "2.11"
+options:
+  argument_spec:
+    description:
+        - A dictionary like AnsibleModule argument_spec
```

Comment:
this should really go into detail using suboptions to describe the spec

Reply:
The spec is described in good detail in the .rst file in this change. I think we should have 1 source describing the spec, otherwise they will go out of sync for certain.

Reply:
why i would keep in in the module (closer to the actual code) as any updates will come here naturally but not to the docs page (which can just point at module). Also would help with arg_spec/docs validation.

Reply:
but not a blocker for this PR, can be addressed afterwards

Reply:
👍